File: lib/presentation/widgets/sections/services_section.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../../../core/constants/app_colors.dart';

class ServicesSection extends StatelessWidget {
  const ServicesSection({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final screenSize = MediaQuery.of(context).size;
    final isMobile = screenSize.width < 768;
    final isTablet = screenSize.width >= 768 && screenSize.width < 1024;

    return Container(
      color: AppColors.background,
      padding: EdgeInsets.symmetric(
        vertical: isMobile ? 56 : 96,
        horizontal: 24,
      ),
      child: Center(
        child: Container(
          constraints: const BoxConstraints(maxWidth: 1200),
          child: Column(
            children: [
              _buildSectionHeader(context, isMobile),
              SizedBox(height: isMobile ? 40 : 64),
              _buildServiceCards(context, isMobile, isTablet),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildSectionHeader(BuildContext context, bool isMobile) {
    return Column(
      children: [
        Text(
          'Everything You Need for Graduate School Success',
          style: TextStyle(
            fontSize: isMobile ? 28 : 36,
            fontWeight: FontWeight.w800,
            color: AppColors.textPrimary,
            fontFamily: 'Inter',
          ),
          textAlign: TextAlign.center,
        ),
      ],
    );
  }

  Widget _buildServiceCards(BuildContext context, bool isMobile, bool isTablet) {
    final services = [
      {
        'emoji': '🔍',
        'title': 'Search Lab Reviews',
        'description': 'Discover authentic reviews from current and former graduate students about research labs, professors, and programs.',
        'features': [
          'Filter by university, department, or research area',
          'Read detailed anonymous reviews',
          'Compare lab ratings and metrics',
          'Save labs to your watchlist',
        ],
        'buttonText': 'Explore Reviews',
        'imagePath': 'images/review_image.png',
        'route': '/reviews',
      },
      {
        'emoji': '📄',
        'title': 'CV & Resume Feedback',
        'description': 'Get professional feedback on your academic CV and resume from experienced graduate students and industry professionals.',
        'features': [
          'AI-powered initial screening',
          'Human expert review and comments',
          'Field-specific formatting guidelines',
          'Before/after improvement tracking',
        ],
        'buttonText': 'Upload Document',
        'imagePath': 'images/resume_image.png',
        'route': '/services/cv-review',
      },
      {
        'emoji': '🎤',
        'title': 'Mock Interview Sessions',
        'description': 'Practice PhD admissions and job interviews with personalized mock sessions tailored to your field and target programs.',
        'features': [
          'AI-powered interview simulation',
          'Field-specific question databases',
          'Real-time feedback and scoring',
          'Video analysis and improvement tips',
        ],
        'buttonText': 'Start Practice',
        'imagePath': 'images/interview_image.png',
        'route': '/services/mock-interview',
      },
    ];

    if (isMobile) {
      return Column(
        children: services.map((service) => Container(
          margin: const EdgeInsets.only(bottom: 24),
          child: _buildServiceCard(
            context: context,
            emoji: service['emoji'] as String,
            title: service['title'] as String,
            description: service['description'] as String,
            features: service['features'] as List<String>,
            buttonText: service['buttonText'] as String,
            imagePath: service['imagePath'] as String,
            route: service['route'] as String,
          ),
        )).toList(),
      );
    } else if (isTablet) {
      return Column(
        children: [
          IntrinsicHeight(
            child: Row(
              children: [
                Expanded(
                  child: _buildServiceCard(
                    context: context,
                    emoji: services[0]['emoji'] as String,
                    title: services[0]['title'] as String,
                    description: services[0]['description'] as String,
                    features: services[0]['features'] as List<String>,
                    buttonText: services[0]['buttonText'] as String,
                    imagePath: services[0]['imagePath'] as String,
                    route: services[0]['route'] as String,
                  ),
                ),
                const SizedBox(width: 24),
                Expanded(
                  child: _buildServiceCard(
                    context: context,
                    emoji: services[1]['emoji'] as String,
                    title: services[1]['title'] as String,
                    description: services[1]['description'] as String,
                    features: services[1]['features'] as List<String>,
                    buttonText: services[1]['buttonText'] as String,
                    imagePath: services[1]['imagePath'] as String,
                    route: services[1]['route'] as String,
                  ),
                ),
              ],
            ),
          ),
          const SizedBox(height: 24),
          Center(
            child: Container(
              constraints: const BoxConstraints(maxWidth: 400),
              child: _buildServiceCard(
                context: context,
                emoji: services[2]['emoji'] as String,
                title: services[2]['title'] as String,
                description: services[2]['description'] as String,
                features: services[2]['features'] as List<String>,
                buttonText: services[2]['buttonText'] as String,
                imagePath: services[2]['imagePath'] as String,
                route: services[2]['route'] as String,
              ),
            ),
          ),
        ],
      );
    } else {
      return IntrinsicHeight(
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: services.map((service) => Expanded(
            child: Container(
              margin: const EdgeInsets.symmetric(horizontal: 12),
              child: _buildServiceCard(
                context: context,
                emoji: service['emoji'] as String,
                title: service['title'] as String,
                description: service['description'] as String,
                features: service['features'] as List<String>,
                buttonText: service['buttonText'] as String,
                imagePath: service['imagePath'] as String,
                route: service['route'] as String,
              ),
            ),
          )).toList(),
        ),
      );
    }
  }

  Widget _buildServiceCard({
    required BuildContext context,
    required String emoji,
    required String title,
    required String description,
    required List<String> features,
    required String buttonText,
    required String imagePath,
    required String route,
  }) {
    return MouseRegion(
      cursor: SystemMouseCursors.click,
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 200),
        child: Material(
          color: Colors.transparent,
          child: InkWell(
            onTap: () => context.go(route),
            borderRadius: BorderRadius.circular(12),
            child: Container(
              decoration: BoxDecoration(
                color: AppColors.surface,
                borderRadius: BorderRadius.circular(12),
                boxShadow: AppColors.cardShadowNarrow,
              ),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                mainAxisSize: MainAxisSize.min,
                children: [
                  // Image thumbnail
                  Container(
                    height: 208,
                    width: double.infinity,
                    decoration: BoxDecoration(
                      color: Colors.grey[200],
                      borderRadius: const BorderRadius.only(
                        topLeft: Radius.circular(12),
                        topRight: Radius.circular(12),
                      ),
                    ),
                    child: ClipRRect(
                      borderRadius: const BorderRadius.only(
                        topLeft: Radius.circular(12),
                        topRight: Radius.circular(12),
                      ),
                      child: Image.asset(
                        imagePath,
                        fit: BoxFit.cover,
                        errorBuilder: (context, error, stackTrace) {
                          // Fallback placeholder with icon
                          return Container(
                            color: AppColors.primary.withOpacity(0.1),
                            child: Column(
                              mainAxisAlignment: MainAxisAlignment.center,
                              children: [
                                Icon(
                                  _getServiceIcon(title),
                                  size: 48,
                                  color: AppColors.primary,
                                ),
                                const SizedBox(height: 8),
                                Text(
                                  title,
                                  style: TextStyle(
                                    color: AppColors.primary,
                                    fontSize: 14,
                                    fontWeight: FontWeight.w600,
                                  ),
                                  textAlign: TextAlign.center,
                                ),
                              ],
                            ),
                          );
                        },
                      ),
                    ),
                  ),

                  // Card content
                  Flexible(
                    child: Padding(
                      padding: const EdgeInsets.all(20),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
                        children: [
                          Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Row(
                                children: [
                                  Text(
                                    emoji,
                                    style: const TextStyle(
                                      fontSize: 20,
                                    ),
                                  ),
                                  const SizedBox(width: 8),
                                  Expanded(
                                    child: Text(
                                      title,
                                      style: const TextStyle(
                                        fontSize: 18,
                                        fontWeight: FontWeight.w700,
                                        color: AppColors.textPrimary,
                                        fontFamily: 'Inter',
                                      ),
                                    ),
                                  ),
                                ],
                              ),
                              const SizedBox(height: 12),
                              Text(
                                description,
                                style: const TextStyle(
                                  fontSize: 15,
                                  color: AppColors.textSecondary,
                                  height: 1.5,
                                  fontFamily: 'Inter',
                                ),
                              ),
                              const SizedBox(height: 16),
                              ...features.map((feature) => Padding(
                                padding: const EdgeInsets.only(bottom: 8),
                                child: Row(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    Container(
                                      width: 16,
                                      height: 16,
                                      margin: EdgeInsets.only(right: 8, top: 2),
                                      decoration: BoxDecoration(
                                        color: Colors.green,
                                        shape: BoxShape.circle,
                                      ),
                                      child: Icon(
                                        Icons.check,
                                        color: Colors.white,
                                        size: 10,
                                      ),
                                    ),
                                    Expanded(
                                      child: Text(
                                        feature,
                                        style: const TextStyle(
                                          fontSize: 14,
                                          color: AppColors.textSecondary,
                                          height: 1.4,
                                          fontFamily: 'Inter',
                                        ),
                                      ),
                                    ),
                                  ],
                                ),
                              )).toList(),
                            ],
                          ),
                          Padding(
                            padding: const EdgeInsets.only(top: 20),
                            child: SizedBox(
                              width: double.infinity,
                              child: ElevatedButton(
                                onPressed: () => context.go(route),
                                style: ElevatedButton.styleFrom(
                                  backgroundColor: AppColors.primary,
                                  foregroundColor: Colors.white,
                                  padding: const EdgeInsets.symmetric(vertical: 12),
                                  shape: RoundedRectangleBorder(
                                    borderRadius: BorderRadius.circular(8),
                                  ),
                                  elevation: 0,
                                ),
                                child: Text(
                                  buttonText,
                                  style: const TextStyle(
                                    fontSize: 14,
                                    fontWeight: FontWeight.w600,
                                    fontFamily: 'Inter',
                                  ),
                                ),
                              ),
                            ),
                          ),
                        ],
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }

  IconData _getServiceIcon(String title) {
    switch (title) {
      case 'Search Lab Reviews':
        return Icons.search;
      case 'CV & Resume Feedback':
        return Icons.description;
      case 'Mock Interview Sessions':
        return Icons.video_call;
      default:
        return Icons.school;
    }
  }
}

File: src/components/ServicesSection.js
import React from 'react';
import { Link } from 'react-router-dom';
import { Search, FileText, Video, Check } from 'lucide-react';
import { colors, shadows, textStyles, spacing, sectionSpacing, borderRadius } from '../theme';

const ServicesSection = () => {
  const screenWidth = window.innerWidth;
  const isMobile = screenWidth < 768;
  const isTablet = screenWidth >= 768 && screenWidth < 1024;

  const services = [
    {
      emoji: '🔍',
      title: 'Search Lab Reviews',
      description: 'Discover authentic reviews from current and former graduate students about research labs, professors, and programs.',
      features: [
        'Filter by university, department, or research area',
        'Read detailed anonymous reviews',
        'Compare lab ratings and metrics',
        'Save labs to your watchlist',
      ],
      buttonText: 'Explore Reviews',
      imagePath: '/assets/images/review_image.png',
      route: '/search',
      icon: Search,
    },
    {
      emoji: '📄',
      title: 'CV & Resume Feedback',
      description: 'Get professional feedback on your academic CV and resume from experienced graduate students and industry professionals.',
      features: [
        'AI-powered initial screening',
        'Human expert review and comments',
        'Field-specific formatting guidelines',
        'Before/after improvement tracking',
      ],
      buttonText: 'Upload Document',
      imagePath: '/assets/images/resume_image.png',
      route: '/services/cv-review',
      icon: FileText,
    },
    {
      emoji: '🎤',
      title: 'Mock Interview Sessions',
      description: 'Practice PhD admissions and job interviews with personalized mock sessions tailored to your field and target programs.',
      features: [
        'AI-powered interview simulation',
        'Field-specific question databases',
        'Real-time feedback and scoring',
        'Video analysis and improvement tips',
      ],
      buttonText: 'Start Practice',
      imagePath: '/assets/images/interview_image.png',
      route: '/services/mock-interview',
      icon: Video,
    },
  ];

  const renderServices = () => {
    if (isMobile) {
      return (
        <div style={{ display: 'flex', flexDirection: 'column', gap: spacing[6] }}>
          {services.map((service, index) => (
            <ServiceCard key={index} service={service} />
          ))}
        </div>
      );
    } else if (isTablet) {
      return (
        <div style={{ display: 'flex', flexDirection: 'column', gap: spacing[6] }}>
          <div style={{ display: 'flex', gap: spacing[6] }}>
            <div style={{ flex: 1 }}>
              <ServiceCard service={services[0]} />
            </div>
            <div style={{ flex: 1 }}>
              <ServiceCard service={services[1]} />
            </div>
          </div>
          <div style={{ display: 'flex', justifyContent: 'center' }}>
            <div style={{ maxWidth: '400px', width: '100%' }}>
              <ServiceCard service={services[2]} />
            </div>
          </div>
        </div>
      );
    } else {
      return (
        <div style={{ display: 'grid', gridTemplateColumns: 'repeat(3, 1fr)', gap: spacing[6] }}>
          {services.map((service, index) => (
            <ServiceCard key={index} service={service} />
          ))}
        </div>
      );
    }
  };

  return (
    <section
      style={{
        backgroundColor: colors.background,
        padding: `${sectionSpacing[isMobile ? 'mobile' : 'desktop'].medium} ${sectionSpacing.horizontal[isMobile ? 'mobile' : 'desktop']}`,
      }}
    >
      <div
        style={{
          maxWidth: '1200px',
          margin: '0 auto',
        }}
      >
        {/* Section Header */}
        <div style={{ textAlign: 'center', marginBottom: isMobile ? spacing[10] : spacing[16] }}>
          <h2
            style={{
              ...textStyles[isMobile ? 'sectionTitleMobile' : 'sectionTitle'],
              color: colors.textPrimary,
            }}
          >
            Everything You Need for Graduate School Success
          </h2>
        </div>

        {/* Service Cards */}
        {renderServices()}
      </div>
    </section>
  );
};

const ServiceCard = ({ service }) => {
  const { emoji, title, description, features, buttonText, imagePath, route, icon: Icon } = service;

  return (
    <Link
      to={route}
      style={{
        textDecoration: 'none',
        color: 'inherit',
        display: 'block',
        height: '100%',
      }}
    >
      <div
        style={{
          backgroundColor: colors.surface,
          borderRadius: borderRadius.md,
          boxShadow: shadows.cardShadowNarrow,
          overflow: 'hidden',
          height: '100%',
          display: 'flex',
          flexDirection: 'column',
          transition: 'all 0.2s ease',
          cursor: 'pointer',
        }}
        onMouseEnter={(e) => {
          e.currentTarget.style.transform = 'translateY(-4px)';
          e.currentTarget.style.boxShadow = shadows.cardShadowHover;
        }}
        onMouseLeave={(e) => {
          e.currentTarget.style.transform = 'translateY(0)';
          e.currentTarget.style.boxShadow = shadows.cardShadowNarrow;
        }}
      >
        {/* Image Thumbnail */}
        <div
          style={{
            height: '208px',
            width: '100%',
            backgroundColor: '#f3f4f6',
            position: 'relative',
            overflow: 'hidden',
          }}
        >
          <img
            src={imagePath}
            alt={title}
            style={{
              width: '100%',
              height: '100%',
              objectFit: 'cover',
            }}
            onError={(e) => {
              // Fallback placeholder with icon
              e.target.style.display = 'none';
              e.target.parentNode.innerHTML = `
                <div style="
                  width: 100%;
                  height: 100%;
                  background-color: rgba(37, 99, 235, 0.1);
                  display: flex;
                  flex-direction: column;
                  align-items: center;
                  justify-content: center;
                  color: ${colors.primary};
                ">
                  <div style="font-size: 48px; margin-bottom: 8px;">${emoji}</div>
                  <div style="font-size: 14px; font-weight: 600; text-align: center;">${title}</div>
                </div>
              `;
            }}
          />
        </div>

        {/* Card Content */}
        <div
          style={{
            padding: spacing[5],
            flex: 1,
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'space-between',
          }}
        >
          <div>
            {/* Title with Emoji */}
            <div style={{ display: 'flex', alignItems: 'center', marginBottom: spacing[3] }}>
              <span style={{ fontSize: '20px', marginRight: spacing[2] }}>{emoji}</span>
              <h3
                style={{
                  ...textStyles.cardTitle,
                  color: colors.textPrimary,
                  flex: 1,
                }}
              >
                {title}
              </h3>
            </div>

            {/* Description */}
            <p
              style={{
                ...textStyles.cardDescription,
                color: colors.textSecondary,
                marginBottom: spacing[4],
                lineHeight: 1.5,
              }}
            >
              {description}
            </p>

            {/* Features */}
            <div style={{ marginBottom: spacing[5] }}>
              {features.map((feature, index) => (
                <div
                  key={index}
                  style={{
                    display: 'flex',
                    alignItems: 'flex-start',
                    marginBottom: spacing[2],
                  }}
                >
                  <div
                    style={{
                      width: '16px',
                      height: '16px',
                      backgroundColor: colors.success,
                      borderRadius: '50%',
                      display: 'flex',
                      alignItems: 'center',
                      justifyContent: 'center',
                      marginRight: spacing[2],
                      marginTop: '2px',
                      flexShrink: 0,
                    }}
                  >
                    <Check size={10} color="white" />
                  </div>
                  <span
                    style={{
                      fontSize: '14px',
                      color: colors.textSecondary,
                      lineHeight: 1.4,
                    }}
                  >
                    {feature}
                  </span>
                </div>
              ))}
            </div>
          </div>

          {/* Button */}
          <button
            style={{
              ...textStyles.buttonTextSmall,
              width: '100%',
              padding: `${spacing[3]} ${spacing[4]}`,
              backgroundColor: colors.primary,
              color: colors.buttonText,
              border: 'none',
              borderRadius: borderRadius.base,
              cursor: 'pointer',
              transition: 'background-color 0.2s ease',
            }}
            onMouseEnter={(e) => {
              e.target.style.backgroundColor = colors.primaryHover;
            }}
            onMouseLeave={(e) => {
              e.target.style.backgroundColor = colors.primary;
            }}
          >
            {buttonText}
          </button>
        </div>
      </div>
    </Link>
  );
};

export default ServicesSection;